### ISSUE TITLE
Field presence in non-message type fields

### ISSUE THREAD

**Issue by chasezheng** [tradewelltech/protarrow#68]
I really love this library. It's very handy. 

For proto3, it's possible for check for presence of a string field. For example,
```proto
message MyMsg {
  optional string value= 1;
}
```
And then we can check
```python
MyMsg.DESCRIPTOR.fields[0].has_presence  # is true

MyMsg().HasField('value')   # is false
```

However, when converting the proto to arrow and then back. The presence bit is not preserved. It would be great if you can change this.

**Comment by 0x26res** [comment]
I think when we started working on protobuf the `optional` label wasn't working as intended.

So instead we used `wrapped` primitive, like `DoubleValue`, `Int32Value` etc. 

I had a look and the current version of protobuf `optional` behave similarly to the wrapped primitive, with less nesting (which is a better API). So we should add support for it.

**Comment by 0x26res** [comment]
@chasezheng this is fixed in v0.4.0. Please give it a try.
